ViewEnvironmentUI/Sources/ViewEnvironmentPropagating.swift

```diff
@@ -351,17 +380,21 @@ public final class ViewEnvironmentUpdateObservationLifetime {
     /// This is called in `deinit`.
     ///
     public func remove() {
+        guard let onRemove else {
+            preconditionFailure("Environment customization was already removed")
```

Comment:
this seems like an assert warning about unexpected behavior may be more appropriate. this is idempotent, right (other than the precondition)? will something break horribly if it's called more than once?

Reply:
Nothing would break if this was called more than once with the current implementation.

I'm fine with moving this to an assert with early return, but I'll check with @watt who suggested this change!

- [ ] Use `assert` instead of `preconditionFailure`?

Reply:
Asserting is fine with me. I think the idea here is that we consider it a programmer error to try to remove a customization more than once, even if it is effectively idempotent.

Reply:
Oh whoops, I added this to the wrong lifetime 🤦 

- [ ] Apply removal assertion changes to customization lifetime